File: fckedu-backend/src/main/java/com/fckedu/exam_creation/question/domain/entity/QuestionEntity.java
package com.fckedu.exam_creation.question.domain.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class QuestionEntity {
    private String id;
    private String categoryId;
    private String lessonId;
    private String exerciseType;
    private String difficultyLevel;
    private List<String> learningOutcomes;
    private String questionType;

    @Builder.Default
    private QuestionContentEntity question = new QuestionContentEntity();

    @Builder.Default
    private List<OptionDataEntity> options = new ArrayList<>();
}

File: frontend/domain/entities/question.entity.ts
export interface QuestionContent {
  template: string;
  variables: {
    math: Record<string, string>;
    image: Record<string, string>;
  };
}

export interface OptionsData {
  template: string;
  variables: {
    math: Record<string, string>;
    image: Record<string, string>;
  };
}

export interface QuestionEntity {
  id: string;
  chapter: string;
  lesson: string;
  exerciseType: string;
  difficultyLevel: string;
  learningOutcomes: string[];
  questionType: string;
  question: QuestionContent;
  options: OptionsData[];
}
